pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -261,7 +261,11 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(proxy *model.Proxy, cl
 	clusterName := util.BuildInboundSubsetKey(proxy, instance.ServicePort.Name,
 		instance.Service.Hostname, instance.ServicePort.Port, clusterPort)
 	localityLbEndpoints := buildInboundLocalityLbEndpoints(bind, instance.Endpoint.EndpointPort)
-	localCluster := cb.buildDefaultCluster(clusterName, cluster.Cluster_STATIC, localityLbEndpoints,
+	clusterType := cluster.Cluster_ORIGINAL_DST
+	if len(localityLbEndpoints) > 0 {
+		clusterType = cluster.Cluster_STATIC
+	}
+	localCluster := cb.buildDefaultCluster(clusterName, clusterType, localityLbEndpoints,
```

Comment:
Should we add higher cleanup interval for this original DST to avoid hosts cleanedup often by Envoy as I mentioned in one of the comments https://docs.google.com/document/d/1j-5_XpeMTnT9mV_8dbSOeU7rfH-5YNtN_JJFZ2mmQ_w/edit#? May not be a big deal but just a thought

Reply:
I set it to 60s now.